llvm/docs/HowToCrossCompileLLVM.rst

```diff
@@ -6,210 +6,202 @@ Introduction
+canonicalization logic at all, meaning for some system headers with symlinks
+in the paths, it can incorrectly compute a non-existing path and consider it
+as always dirty. If you are suffering from this issue, you will find at any
+attempt at an incremental build (including the suggested command to build the
+``install`` target in the next section) results in recompiling everything.
+``ninja -C build/$TARGET -t deps`` shows files in ``$SYSROOT/include/*`` that
+don't (as the ``$SYSROOT/include`` folder doesn't exist) and you can
+further confirm these files are causing the build to be seen as dirty with
+``ninja -C build/$TARGET -d deps``. Until such time as the issue is avoided
+due to a change in behaviour for Clang or for Ninja, a functional workaround
+is to create a symlink so that the incorrect ``$SYSROOT/include/*``
+dependencies resolve to files within ``$SYSROOT/usr/include/*``. This works in
+practice for the simple cross-compilation use case described here, but isn't
+a general solution.
 
-It should automatically find out how many cores you have, what are
-the rules that needs building and will build the whole thing.
+   .. code-block:: bash
 
-You can't run ``ninja check-all`` on this tree because the created
-binaries are targeted to ARM, not x86_64.
+    sudo ln -s usr/include $SYSROOT/include
 
-Installing and Using
+Installing and using
 --------------------
 
-After the LLVM/Clang has built successfully, you should install it
-via:
-
-   .. code-block:: bash
-
-     $ ninja install
-
-which will create a sysroot on the install-dir. You can then tar
-that directory into a binary with the full triple name (for easy
-identification), like:
+After LLVM/Clang has built successfully, you can install it via:
 
    .. code-block:: bash
 
-     $ ln -sf <install-dir> arm-linux-gnueabihf-clang
-     $ tar zchf arm-linux-gnueabihf-clang.tar.gz arm-linux-gnueabihf-clang
+    cmake --build build/$TARGET --target=install
 
-If you copy that tarball to your target board, you'll be able to use
-it for running the test-suite, for example. Follow the guidelines at
-https://llvm.org/docs/lnt/quickstart.html, unpack the tarball in the
-test directory, and use options:
+This will install the LLVM/Clang headers, binaries, libraries, and other files
+to paths within ``CMAKE_INSTALL_PREFIX`` You can then tar that directory for
+transfer to a device that runs the target architecture natively:
 
    .. code-block:: bash
 
-     $ ./sandbox/bin/python sandbox/bin/lnt runtest nt \
-         --sandbox sandbox \
-         --test-suite `pwd`/test-suite \
-         --cc `pwd`/arm-linux-gnueabihf-clang/bin/clang \
-         --cxx `pwd`/arm-linux-gnueabihf-clang/bin/clang++
-
-Remember to add the ``-jN`` options to ``lnt`` to the number of CPUs
-on your board. Also, the path to your clang has to be absolute, so
-you'll need the `pwd` trick above.
+    tar -czvf clang-$TARGET.tar.gz -C $HOME clang-$TARGET
```

Comment:
It is implicit here that this toolchain is portable, should we make it explicit? (as much as we can anyway, if you have used a Debian sysroot, it's not going to do well on another distro maybe)

Reply:
I've added a comment on this, please take a look. I don't want to get too into the weeds on compatibility of library versions etc so the comment "This means using a sysroot that is as
similar to your target operating system as possible is desirable." doesn't quite tell the whole story. Using a sysroot for an older distro version can be a workable strategy to maximise compatibility. But I don't think it's a bad suggestion for people to follow who don't have a good reason not to. Thought I'd flag it for consideration though.

Reply:
Yeah that's fine. I was mostly thinking about literal file paths but yes library versioning is also a thing.

The answer to actual file paths is that it is as portable as it can be, partly because we did that absolute to relative symlink thing earlier.